core/src/main/scala/reftree/diagram/Animation.scala

```diff
@@ -119,6 +107,35 @@ object Animation {
     def iterate(n: Int)(iteration: A ⇒ A) =
       Builder((1 to n).foldLeft(frames)((current, _) ⇒ current :+ iteration(current.last)))
 
+    /** Add more frames by applying the provided iteration function until `predicate` yields true */
+    def iterateUntilAtMost(n: Int)(iteration: (A => A))(predicate: (A => Boolean)) =
+      if (frames.isEmpty) this
```

Comment:
I suggest putting `predicate` before `iteration` (I think I elaborated in some other comment). By the way, could you remove the redundant parentheses around function types (would be awesome if you could also use the unicode arrow, but I can amend that later myself if you prefer).